sys/kern/uipc_shm.c

```diff
@@ -1007,6 +1022,30 @@ shm_drop(struct shmfd *shmfd)
 	}
 }
 
+void
+shm_vmspace_free(struct vmspace *vm)
+{
+	struct shmfd *shmfd;
+
+	/*
+	 * XXX-BD: do we actually need a lock given all references have
```

Comment:
I think you do need a lock. shm_drop() might be executed by a different process, and they need to synchronize somehow.